### PR TITLE
Field guide refactoring

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-classifier

Describe your changes:
I took the opportunity to refactor the field guide components to use the newer mobx-react recommendation of using the React Context API. This means removing the older inject decorators wherever they were used and adding a connector component for the store connection in a pattern similar to what I used with the variable star light curve viewer. In addition, I removed the redundant `SpacedHeading` since that now exists in the component library and did some minor file reorganization. The impacted components have updated tests.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
@ErikOstlund this PR is part of a larger bit of work that I'm doing to refactor the field guide. This PR is scoped to the mobx-react changes I wanted to make. Let me know if you have questions and if you want to do a screen share to review it together next week. Thanks!

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideContainer.js [hunk 1]
```diff
@@ -1,44 +1,47 @@
...
+FieldGuideContainer.defaultProps = {
+  setActiveItemIndex: -1,
+  icons: observable.map(),
   showModal: false
 }
 
-FieldGuideContainer.wrappedComponent.propTypes = {
+FieldGuideContainer.propTypes = {
+  activeItemIndex: PropTypes.number,
+  fieldGuide: PropTypes.object.isRequired,
```
Comment (ErikOstlund) [inline packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideContainer.js:42]: Make `fieldGuide` optional.

**Comment by srallen** [comment]
@ErikOstlund I've made the requested change and this can be tested again. 

Project without a field guide
  - https://localhost:8080/ (default kitty test project)
  -  Button is disabled
An in production project with field guide
  - https://localhost:8080/?project=4878&workflow=16135&env=production
  - button is enabled, clickable, and opens the guide on the right side of the screen.

**Review by ErikOstlund — APPROVED** [review]
All unit tests pass. Field guide appears and functions as expected. Looks good! 👍🏽

**Pull request merged**